src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java

```diff
@@ -688,6 +714,271 @@ public boolean apply(@Nullable FlowNode input) {
         });
     }
 
+    @Issue("JENKINS-36547")
+    @Test public void reuseNodeFromPreviousRun() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                for (int i = 0; i < 5; ++i) {
+                    DumbSlave slave = story.j.createOnlineSlave();
+                    slave.setLabelString("foo bar");
+                }
+
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "demo");
+                p.setDefinition(new CpsFlowDefinition("node('foo') {\n" +
+                        "}\n", true));
+
+                WorkflowRun run = story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+                List<WorkspaceAction> workspaceActions = getWorkspaceActions(run);
+                assertEquals(workspaceActions.size(), 1);
+
+                String firstNode = workspaceActions.get(0).getNode();
+                assertNotEquals(firstNode, "");
+
+                WorkflowRun run2 = story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+                workspaceActions = getWorkspaceActions(run2);
+                assertEquals(workspaceActions.size(), 1);
+                assertEquals(workspaceActions.get(0).getNode(), firstNode);
+            }
+        });
+    }
+
+    private Map<String, String> mapNodeNameToLogText(WorkflowRun workflowRun) throws java.io.IOException{
+        FlowGraphWalker walker = new FlowGraphWalker(workflowRun.getExecution());
+        Map<String, String> nodeNameToLogText = new HashMap<>();
+        StringBuilder previousLogText = new StringBuilder();
+        for (FlowNode n : walker) {
+            WorkspaceAction a = n.getAction(WorkspaceAction.class);
+            LogActionImpl l = n.getAction(LogActionImpl.class);
```

Comment:
This is the old way of storing log data, so now that the workflow-job plugin used in testing is version 2.26 (which switched to `LogStorageAction`), the log text for each node is always just `""`, which seems unintentional. I think you should replace it with `LogAction.class` to avoid depending on the specific implementation.

Reply:
Thanks, I'll adjust this accordingly.